magda-storage-api/src/MagdaMinioClient.ts

```diff
@@ -17,26 +25,41 @@ export default class MagdaMinioClient implements ObjectStoreClient {
             secretKey,
             region
         });
-        this.client.makeBucket(this.bucketName, region, (err: Error) => {
-            if (err) {
-                if (
-                    err.message ===
-                    "Your previous request to create the named bucket succeeded and you already own it."
-                ) {
-                    return console.log(
-                        "Bucket " + this.bucketName + " already exists 👍"
-                    );
-                } else {
-                    return console.log("😢 Error creating bucket: ", err);
+        this.region = region;
+        this.createBucket(this.bucketName);
+    }
+
+    createBucket(bucket: string): Promise<any> {
```

Comment:
Looks like it returns a consistent object shape, could we make that the return type of the `Promise`? :)

Reply:
Yep, created the type `CreateBucketResponse` to reflect that.